includes/admin/class-wc-rest-upe-flag-toggle-controller.php

```diff
@@ -121,6 +121,10 @@ private function update_is_upe_enabled( WP_REST_Request $request ) {
 			return;
 		}
 
+		// marking the flag as "disabled", so that we can keep track that the merchant explicitly disabled it.
+		update_option( WC_Payments_Features::UPE_FLAG_NAME, 'disabled' );
```

Comment:
This change makes `_wcpay_feature_upe` to have 3 states, 0 (disabled, aka not-yet-enabled), 1, 'disabled' (disabled, user clicked disabled). 

One thing to note here is there are a few places that use `WC_Payments_Features::is_upe_enabled()`. If `_wcpay_feature_upe = 'disabled'`, then all of those places would treat it as `false`; which I think is okay. "Disabled" would imply "not enabled". It would have the same effect as `0`.

Vice versa, `WC_Payments_Features::did_merchant_disable_upe()` would return false if `_wcpay_feature_upe` is `1` or `0`, which also seems to be okay. 

It seems like `0`, `1`, `disabled` are mutually exclusive. I think this is fine.

Reply:
Yeah, those were my thoughts as well.
I wanted to avoid creating a new entry in `wp_options` to track whether the feature had been manually disabled to hide the note/onboarding task.

Reply:
Yea i think less feature flag is a good idea. Especially if they can be combined into the same one.